packages/jellyfish-transaction/src/script/defi/dftx_governance.ts

```diff
@@ -0,0 +1,91 @@
+import BigNumber from 'bignumber.js'
+import { SmartBuffer } from 'smart-buffer'
+import { readBigNumberUInt64, writeBigNumberUInt64 } from '../../buffer/buffer_bignumber'
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export interface LiqPoolSplit {
+  tokenId: number // -------------------| 4 bytes unsigned
+  value: BigNumber // ------------------| 8 bytes unsigned
+}
+
+export class CLiqPoolSplit extends ComposableBuffer<LiqPoolSplit> {
+  composers (lps: LiqPoolSplit): BufferComposer[] {
+    return [
+      ComposableBuffer.uInt32(() => lps.tokenId, v => lps.tokenId = v),
+      ComposableBuffer.satoshiAsBigNumber(() => lps.value, v => lps.value = v)
+    ]
+  }
+}
+
+export interface GovernanceVar {
+  key: string // -----------------------| VarUInt{1-9 bytes}
+  value: BigNumber | LiqPoolSplit[] // -| VarUInt{8 OR 1 + n * 12 bytes}, case LiqPoolSplit: first byte = array len
```

Comment:
Have to manual check and cast value type after "key" decoded.

Reply:
We could refactor into something like this:

```ts
export interface GovernanceLpSplits {
  key: 'LP_SPLITS'
  splits: LiqPoolSplit[]
}
```

and 

```ts
export interface GovernanceLpDailyReward {
  key: 'LP_DAILY_DFI_REWARD'
  reward: BigNumber
}
```

This way it will be resilient against changes in the future.

and 

```ts
export interface GovernanceUnmapped {
  key: string
  hex: string
}
```